internal/bzlmod/non_module_deps.bzl

```diff
@@ -28,6 +29,28 @@ def _non_module_deps_impl(module_ctx):
         config = Label("@bazel_gazelle_go_repository_directives//:WORKSPACE"),
     )
 
+    # Required when depending on org_golang_google_grpc. The patches are
+    # pulled from @rules_go
+    http_archive(
+        name = "go_googleapis",
```

Comment:
I got stuck on wanting to integrate this troublesome repo in a better way, but it's true that we don't have to couple this to the bzlmod migration.